### PR TITLE
Remove TODO and implemented code version snapshot in ER creation

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java [hunk 1]
```diff
@@ -630,7 +632,15 @@ public FutureExperimentRunDAO(Executor executor, FutureJdbi jdbi, UAC uac) {
...
-        // TODO .thenCompose(handle -> addCodeVersionSnapShot(), executor)
+        .thenCompose(
+            handle ->
+                codeVersionHandler.logCodeVersion(
+                    LogExperimentRunCodeVersion.newBuilder()
+                        .setId(newExperimentRun.getId())
+                        .setCodeVersion(newExperimentRun.getCodeVersionSnapshot())
+                        .setOverwrite(false)
+                        .build()),
+            executor)
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java]: Same feedback

File: backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CodeVersionHandler.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+import ai.verta.modeldb.utils.RdbmsUtils;
+import java.util.Optional;
+import java.util.concurrent.Executor;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.hibernate.LockMode;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+public class CodeVersionHandler {
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CodeVersionHandler.java]: I have already implemented this class.

**Review by conradoverta — CHANGES_REQUESTED** [review]

File: backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CodeVersionHandler.java (hunk 2 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CodeVersionHandler.java]: Yes, this is extracted from your PR

File: backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java (hunk 1 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java]: Fixed

File: backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CodeVersionHandler.java (hunk 2 above)
Comment (conradoverta) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CodeVersionHandler.java]: Just point your branch to base off mine then. Let's not copy code between branches as it makes things much harder to manage.

**Review by conradoverta — COMMENTED** [review]
Seems like `CodeVersionHandler` is still copied, right?

File: backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CodeVersionHandler.java (hunk 2 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CodeVersionHandler.java]: Updated branch based with your code

**Comment by anandjakhaniya** [comment]
> Seems like `CodeVersionHandler` is still copied, right?

Updated PR right now.

**Review by conradoverta — APPROVED** [review]

**Pull request merged**